### PR TITLE
Add __slots__ to timer helpers

### PR DESCRIPTION
**Pull request by bdraco**
## What do these changes do?

We use `__slots__` almost everywhere else in the codebase, however `__slots__` was missing for these helpers

related issue https://github.com/aio-libs/aiohttp/issues/4618#issuecomment-2391880153

## Are there changes in behavior for the user?

no

## Is it a substantial burden for the maintainers to support this?
no

### CODE REVIEW

**Comment by bdraco** [comment]
https://github.com/python/cpython/pull/106771

File: pyproject.toml [hunk 1]
```diff
@@ -82,3 +82,8 @@ skip = "pp*"
 [tool.codespell]
 skip = '.git,*.pdf,*.svg,Makefile,CONTRIBUTORS.txt,venvs,_build'
 ignore-words-list = 'te,assertIn'
+
+[tool.slotscheck]
+# Remove require-superclass = false once https://github.com/python/cpython/pull/106771
+# is available in all supported cpython versions
```
Comment (bdraco) [inline pyproject.toml:88]: I don't really like this. I think this check is valuable.  It's probably better to exclude helpers.py instead

File: pyproject.toml [hunk 2]
```diff
@@ -82,3 +82,8 @@ skip = "pp*"
 [tool.codespell]
 skip = '.git,*.pdf,*.svg,Makefile,CONTRIBUTORS.txt,venvs,_build'
 ignore-words-list = 'te,assertIn'
+
+[tool.slotscheck]
+# Remove aiohttp.helkpers once https://github.com/python/cpython/pull/106771
```
Comment (bdraco) [inline pyproject.toml]:
```suggestion
# Remove aiohttp.helpers once https://github.com/python/cpython/pull/106771
```

File: pyproject.toml [hunk 3]
```diff
@@ -82,3 +82,8 @@ skip = "pp*"
 [tool.codespell]
 skip = '.git,*.pdf,*.svg,Makefile,CONTRIBUTORS.txt,venvs,_build'
 ignore-words-list = 'te,assertIn'
+
+[tool.slotscheck]
+# Remove aiohttp.helpers once https://github.com/python/cpython/pull/106771
```
Comment (Dreamsorcerer) [inline pyproject.toml]: Appears to be 3.13 minimum.
```suggestion
# TODO(3.13): Remove aiohttp.helpers once https://github.com/python/cpython/pull/106771
```

**Pull request merged**